packages/grafana-data/src/types/live.ts

```diff
@@ -85,28 +85,30 @@ export interface LiveChannelStatus {
   /**
    * The last error.
    *
-   * This will remain in the status until a new message is succesfully recieved from the channel
+   * This will remain in the status until a new message is succesfully received from the channel
    */
   error?: any;
 }
 
 /**
  * @experimental
  */
-export interface LiveChannelJoinLeave {
+export interface LiveChannelPresenseEvent {
+  action: 'join' | 'leave';
   user: any;
 }
 
 /**
  * @experimental
  */
-export interface LiveChannelPresense {
+export interface LiveChannelPresenseStatus {
   users: any;
 }
 
-export interface LiveChannelMessage<TMessage = any> {
-  type: 'status' | 'message' | 'join' | 'leave';
```

Comment:
@hugohaggmark -- I switched from having a type flag to a limited set of properties that may exist.  This works better for typing, and the check for `if (event.status) {` seems more clear as `if (event.type === 'status') {`

Reply:
Hmm, you can easily create TypeScript guard functions for that as in isStatusEvent, isLeaveEvent etc. WDYT?

Reply:
But then you still have to cast message to the types value - or is there a typescript magic way to do that?

The semantic difference is this would now support multiple message types in one event (not great -- bet better than lots of casting)

Reply:
No need for cast when you use a type guard as TypeScript then knows what type it is then, and yes it's magic :)
https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types

Reply:
Updated with a new type structure I think you will like ;)  and OMG that is magic.  I feared you would have to cast after the check